source/MapSalesPanel.cpp

```diff
@@ -266,18 +266,7 @@ void MapSalesPanel::DrawPanel() const
 		Point(WIDTH, Screen::Height()),
 		back);
 
-	const Sprite *edgeSprite = SpriteSet::Get("ui/right edge");
-	if(edgeSprite->Height())
-	{
-		int steps = Screen::Height() / edgeSprite->Height();
-		for(int y = -steps; y <= steps; ++y)
-		{
-			Point pos(
-				Screen::Width() * -.5f + WIDTH + .5f * edgeSprite->Width(),
-				y * edgeSprite->Height());
-			SpriteShader::Draw(edgeSprite, pos);
-		}
-	}
+	Panel::DrawEdgeSprite(SpriteSet::Get("ui/right edge"), Screen::Width() * -.5f + WIDTH);
```

Comment:
Getting a slight gap here for some reason.
![image](https://github.com/endless-sky/endless-sky/assets/17688683/9a81732b-83d5-43e9-aeb7-41e9bbeee3da)
The other locations look fine.

Reply:
Really? I'm not seeing any difference between the 1x and 2x images that could cause this. This gap is only one pixel wide in the screenshot, so might it be that there's an odd dimension in the interface code that rounds this pixel out at certain zoom levels? I get this gap at 110%, 140%, and 170%, but not any of the other zoom factors between 100% and 190% (I can't access 200% on my monitor to test it). Every time the gap exists, it's a single pixel.

Reply:
Actually, I see what it is.

Here's the fill shader from MessagesLogPanel:
```c++
	FillShader::Fill(
		Point(Screen::Left() + .5 * width, 0.),
		Point(width, Screen::Height()),
		backColor);
```
ConversationPanel:
```c++
	FillShader::Fill(
		Point(Screen::Left() + .5 * boxWidth, 0.),
		Point(boxWidth, Screen::Height()),
		back);
```
LogbookPanel:
```c++
	// Draw the panel. The sidebar should be slightly darker than the rest.
	const Color &sideColor = *GameData::Colors().Get("logbook sidebar");
	FillShader::Fill(
		Point(Screen::Left() + .5 * SIDEBAR_WIDTH, 0.),
		Point(SIDEBAR_WIDTH, Screen::Height()),
		sideColor);
	const Color &backColor = *GameData::Colors().Get("logbook background");
	FillShader::Fill(
		Point(Screen::Left() + SIDEBAR_WIDTH + .5 * TEXT_WIDTH, 0.),
		Point(TEXT_WIDTH, Screen::Height()),
		backColor);
	const Color &lineColor = *GameData::Colors().Get("logbook line");
	FillShader::Fill(
		Point(Screen::Left() + SIDEBAR_WIDTH - .5, 0.),
		Point(1., Screen::Height()),
		lineColor);
```
And then the problem child, MapSalesPanel:
```
	FillShader::Fill(
		Point(Screen::Width() * -.5 + WIDTH * .5, 0.),
		Point(WIDTH, Screen::Height()),
		back);
```
MapSalesPanel is using `Screen::Width() * -.5`, whereas the others are all using `Screen::Left()`. If I make that change then the problem goes away.